packages/gitgraph-core/src/user-api/branch-user-api.ts

```diff
@@ -280,8 +280,10 @@ class BranchUserApi<TNode> {
         ...withoutUndefinedKeys(
           this._branch.commitDefaultOptions.style!.message,
         ),
-        display: this._graph.shouldDisplayCommitMessage,
         ...style.message,
+        ...withoutUndefinedKeys({
+          display: this._graph.shouldDisplayCommitMessage && undefined,
```

Comment:
I see what you did there 😉 

I'm fine with that. I'll add a comment to explain so the intent is clear.

Possible alternatives would have been:
- using a temporary variable, as before (more verbose though)
- doing something like `...(!this._graph.shouldDisplayCommitMessage && { display: false })` but honestly, I think this would be obscure too 😁